subprojects/files/src/main/java/org/gradle/api/internal/file/AbstractFileCollection.java

```diff
@@ -72,14 +71,29 @@ public File getSingleFile() throws IllegalStateException {
         return getFiles().iterator();
     }
 
+    @Override
+    public boolean visitContents(CancellableVisitor<File> visitor) {
+        for (File file : getFiles()) {
```

Comment:
Why don't you iterate over `this` instead of over `getFiles()`? Would it be sometimes cheaper to iterate over `this` instead of being forced to create the file collection?

Reply:
That's a very good point.

Reply:
Fixed.